### PR TITLE
Summarization Examples: add Bart CNN Evaluation

### LINKED ISSUE #2847: BART/T5 seq2seq example
# 🚀 Feature request

Can we have a seq2seq example with training/fine-tuning and generation for BART/T5 models?

### LINKED ISSUE #3023: BART : host `bart-large-cnn`
# 🌟 New model addition

BART pretrained model `bart-large` is currently provided, as well as fine-tuned BART model `bart-large-mnli`.

**How about [`bart-large-cnn`](https://github.com/pytorch/fairseq/tree/master/examples/bart#pre-trained-models) ?**

### PR DESCRIPTION
**Pull request by sshleifer**

### CODE REVIEW

File: examples/test_examples.py [hunk 1]
```diff
@@ -98,3 +99,9 @@ def test_generation(self):
         with patch.object(sys, "argv", testargs + [model_type, model_name]):
             result = run_generation.main()
             self.assertGreaterEqual(len(result[0]), 10)
+
+    def test_bart_cnn_summarization(self):
```
Comment (sshleifer) [inline examples/test_examples.py]: takes 15 seconds. Happy to mark it slow or move it somewhere else!

**Review by patrickvonplaten — APPROVED** [review]

**Review by LysandreJik — COMMENTED** [review]
Nice, this is clean! We try to have our examples as model-agnostic as possible (see `run_glue`, `run_squad` or `run_generation`). If we have some model-specific examples, we move them to their own directory (see `pplm/` or `mm-imdb/`).

What would make sense to you? Renaming this `run_summarization.py` (removing the BART part) or move it to `bart/evaluate_bart_cnn.py`?

File: examples/test_examples.py (hunk 1 above)
Comment (julien-c) [reply examples/test_examples.py]: yes let's maybe keep that in slow

Comment (sshleifer) [reply examples/test_examples.py]: Non-trivial to import `slow` from tests.utils, should I reimplement, hack sys.path, move tests/utils to src/transformers or a better idea I haven't thought of?

Comment (julien-c) [reply examples/test_examples.py]: ha no just leave it as non-slow then

**Comment by sshleifer** [comment]
image
New organization @LysandreJik

**Comment by LysandreJik** [comment]
I like that organisation

**Review by LysandreJik — APPROVED** [review]
LGTM

**Pull request merged**